pandas/tests/series/test_dtypes.py

```diff
@@ -487,3 +487,54 @@ def test_reindex_astype_order_consistency(self):
         s1 = s.reindex(new_index).astype(temp_dtype).astype(new_dtype)
         s2 = s.astype(temp_dtype).reindex(new_index).astype(new_dtype)
         tm.assert_series_equal(s1, s2)
+
+    @pytest.mark.parametrize(
+        "series, dtype",
+        [
+            (Series([1, 2, 3], dtype=np.dtype("int32")), "Int32"),
+            (Series([1, 2, 3], dtype=np.dtype("int64")), "Int64"),
+            (Series(["x", "y", "z"], dtype=np.dtype("O")), pd.StringDtype()),
+            (Series([True, False, np.nan], dtype=np.dtype("O")), pd.BooleanDtype()),
+            (Series(["h", "i", np.nan], dtype=np.dtype("O")), pd.StringDtype()),
+            (Series([10, np.nan, 20], dtype=np.dtype("float")), pd.Int64Dtype()),
+            (Series([np.nan, 100.5, 200], dtype=np.dtype("float")), np.dtype("float")),
+            (Series([3, 4, 5], dtype="Int8"), "Int8"),
+            (Series([[1, 2], [3, 4], [5]]), np.dtype("O")),
+            (Series([4, 5, 6], dtype=np.dtype("uint32")), "UInt32"),
+            (Series([-10, 12, 13], dtype=np.dtype("i1")), "Int8"),
+            (Series([1, 2.0], dtype=object), "Int64"),
+            (Series(["a", "b"], dtype=pd.CategoricalDtype()), pd.CategoricalDtype()),
+            (
+                Series(
+                    pd.to_datetime(["2020-01-14 10:00", "2020-01-15 11:11"]),
+                    dtype=pd.DatetimeTZDtype(tz="UTC"),
+                ),
+                pd.DatetimeTZDtype(tz="UTC"),
+            ),
+            (
+                Series(
+                    pd.to_datetime(["2020-01-14 10:00", "2020-01-15 11:11"]),
+                    dtype="datetime64[ns]",
+                ),
+                np.dtype("datetime64[ns]"),
+            ),
+            (
+                Series(pd.period_range("1/1/2011", freq="M", periods=3)),
+                pd.PeriodDtype("M"),
+            ),
+            (
+                Series(pd.arrays.IntervalArray([pd.Interval(0, 1), pd.Interval(1, 5)])),
+                pd.IntervalDtype("int64"),
+            ),
+        ],
+    )
+    def test_as_nullable_dtypes(self, series, dtype):
+        ns = series.as_nullable_dtypes()
+        assert ns.dtype == dtype
+
+        if isinstance(series.dtype, type(ns.dtype)) and series.dtype == ns.dtype:
```

Comment:
huh? what is this trying to do

Reply:
See comment on the line below, it is testing that the result of as_nullable_dtypes is always a copy. 

@Dr-Irv you are testing this now only when the dtype of the result is the same, but I think to simplify the test here, you can simply always do this test? (also when an actual conversion was done, it should be a copy, right?)

Reply:
ok, fixed.